mrp_production_byproduct_cost_share/report/mrp_report_bom_structure.py

```diff
@@ -0,0 +1,258 @@
+    #     data["lines"] = lines
+    #     return data
+
+    def _get_pdf_line(
+        self, bom_id, product_id=False, qty=1, child_bom_ids=[], unfolded=False
+    ):
+        data = super()._get_pdf_line(bom_id, product_id, qty, child_bom_ids, unfolded)
+
+        def get_sub_lines(bom, product_id, line_qty, line_id, level):
+            # method overriden
+            data = self._get_bom(
+                bom_id=bom.id,
+                product_id=product_id,
+                line_qty=line_qty,
+                line_id=line_id,
+                level=level,
+            )
+            bom_lines = data["components"]
+            lines = []
+            for bom_line in bom_lines:
+                lines.append(
+                    {
+                        "name": bom_line["prod_name"],
+                        "type": "bom",
+                        "quantity": bom_line["prod_qty"],
+                        "uom": bom_line["prod_uom"],
+                        "prod_cost": bom_line["prod_cost"],
+                        "bom_cost": bom_line["total"],
+                        "level": bom_line["level"],
+                        "code": bom_line["code"],
+                        "child_bom": bom_line["child_bom"],
+                        "prod_id": bom_line["prod_id"],
+                    }
+                )
+                if bom_line["child_bom"] and (
+                    unfolded or bom_line["child_bom"] in child_bom_ids
+                ):
+                    line = self.env["mrp.bom.line"].browse(bom_line["line_id"])
+                    lines += get_sub_lines(
+                        line.child_bom_id,
+                        line.product_id.id,
+                        bom_line["prod_qty"],
+                        line,
+                        level + 1,
+                    )
+            if data["operations"]:
+                lines.append(
+                    {
+                        "name": _("Operations"),
+                        "type": "operation",
+                        "quantity": data["operations_time"],
+                        "uom": _("minutes"),
+                        "bom_cost": data["operations_cost"],
+                        "level": level,
+                    }
+                )
+                for operation in data["operations"]:
+                    if unfolded or "operation-" + str(bom.id) in child_bom_ids:
+                        lines.append(
+                            {
+                                "name": operation["name"],
+                                "type": "operation",
+                                "quantity": operation["duration_expected"],
+                                "uom": _("minutes"),
+                                "bom_cost": operation["total"],
+                                "level": level + 1,
+                            }
+                        )
+            # start of changes
+            if data["byproducts"]:
+                lines.append(
```

Comment:
I see start of changes and no "end of changes", so I'm assuming this has changes until the end of the method. If so, you can do this extension with normal inheritance, can't you?

```python
def get_sub_lines(...)
    lines = super()
    if data["byproducts"]:
        lines.append(...)
    return lines
```

Reply:
get_sub_lines is a method declared inside _get_pdf_line, so the usual inheritance here didn't work:)

Reply:
ouch, I see. Thanks for clarifying!